addons/pos_loyalty/models/loyalty_reward.py

```diff
@@ -11,3 +11,9 @@ def _get_discount_product_values(self):
         for vals in res:
             vals.update({'taxes_id': False})
         return res
+
+    def unlink(self):
+        if self.env['pos.order.line'].search_count([('reward_id', '=', self.id)]):
```

Comment:
This is not correct when unlink is called with multiple records 

Maybe something like https://github.com/odoo/odoo/commit/cbb1ed990bc1236c1c12c4fb40f7d03437cd7b07#diff-200f47146e57d8411c0824088bdfc6e119cedc836607e6559f4978230d075e55L10